### ISSUE TITLE
read newick format

### ISSUE THREAD

**Issue by mufernando** [grahamgower/demes#82]
It would be cool to be able to import a Newick tree (with or without node ages) to form a base `Deme` object that can then be annotated with other details (split times, population sizes, etc).

Similar to `msprime`s feature: https://msprime.readthedocs.io/en/latest/tutorial.html#parsing-species-trees

I can definitely lend a hand if needed.

**Comment by grahamgower** [comment]
Thanks @mufernando, that seems very doable!

**Comment by jeromekelleher** [comment]
Good idea. One downside though is that we start adding dependencies. We're using the [newick](https://pypi.org/project/newick/) package in msprime which is pretty lightweight, but it doesn't support extended newick, which would ultimately be needed to do this properly.

Still, it seems worth it, and we can mostly just copy the code from msprime I think.

**Comment by grahamgower** [comment]
For any GPL code we want to copy from msprime, we'll have to explicitly contact the original authors to check they're happy to have their code modified and more permissively licensed for inclusion in `demes`.
```
$ git blame species_trees.py -c | cut -f2 | sed 's/(//' | sort -u
Ben Jeffery
Jerome Kelleher
Michael Matschiner
```

**Comment by jeromekelleher** [comment]
I'm confident that we can get this approval.
